R/adapter.r

```diff
@@ -151,9 +151,11 @@ construct_file_adapter <- function() {
     if ('.rds' == substring(opts$resource, nchar(opts$resource) - 3, nchar(opts$resource)))
       readRDS(opts$resource)
     else {
+      test_if_url <- function(str) grepl("^https?://", str)
+      filesource <- ifelse(test_if_url(opts$resource), getURL(opts$resource), opts$resource)
```

Comment:
Doesn't `getURL` retrieve the actual content? That won't work as a `file` key below. Also, `read.csv` uses a fileConnection, which does URL parsing by default. For example, `read.csv("http://therobert.org/stuff/wizzo/iris.csv")` works for me.